### ISSUE TITLE
terra-date-picker focus highlighting issue on IE 

### ISSUE THREAD

**Issue by supreethmr** [cerner/terra-framework#1247]
# Bug Report
Focus highlights only on three sides of date-picker button on IE.

## Description
When date-picker button is focused via keyboard focus ring gets overlapped by input in IE. This shown-up  due to recent change of date-picker style `inline-block` to `inline-flex` which is causing input to overlap the calendar-button in Microsoft browsers.

## Steps to Reproduce

1. Open IE and Navigate to : https://engineering.cerner.com/terra-ui/components/terra-date-picker/date-picker/date-picker#examples
2. click on any date-picker input and press tab to focus calendar button
3. focus appears only on three sides of button

## Additional Context / Screenshots

<img width="251" alt="Screen Shot 2020-09-14 at 6 44 59 PM" src="https://user-images.githubusercontent.com/12869809/93092673-5d224400-f6bd-11ea-80ab-2dff3ba39649.png">

## Expected Behavior

Focus style should remain consistent across the browsers ( IE11, IE 10, Edge )

## Possible Solution

Since input overlaps the button focus styles in IE adding z-index similar to [invalid focus style](https://github.com/cerner/terra-framework/blob/6472407c71f6460c62531127c362ed0622a4551e/packages/terra-date-picker/src/DatePicker.module.scss#L83) would fix this issue.

## Environment

* Component Name and Version:
* Browser Name and Version: IE 11
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile): Windows

**Comment by benbcai** [comment]
@supreethmr I'm only able to see this issue on IE11 using the examples in [terra-framework](https://engineering.cerner.com/terra-framework/components/terra-date-picker/date-picker/date-picker). Using the same example in [terra-ui]([url](https://engineering.cerner.com/terra-ui/components/terra-date-picker/date-picker/date-picker)) on IE11 appears to work for me. Perhaps, the change you mentioned for changing inline-block to inline-flex is not in terra-ui? Which change is this that you're referring to?

**Comment by supreethmr** [comment]
Yes it looks like the changes are not yet updated in terra-ui. I was referring to changes made in PR #1239 : https://github.com/cerner/terra-framework/pull/1239/files#diff-b78b5ee7af98571b0add4166934cac73 here date-picker styles were updated to inline block to fix some alignment issues I guess. due to this input overlaps the button in IE.

**Comment by benbcai** [comment]
It makes sense that terra-ui probably doesn't have what ever change affected this. It is weird that this change didn't get merged until this morning but the team is seeing this issue even before this change is merged.

**Comment by supreethmr** [comment]
💡 That's correct I think that's why it was not reproducible earlier in dev-site examples as well as in local main code. I re-visited the screenshots posted in teams and yes they were having this issue with date-picker with placeholder and not with latest date-picker. But the code changes of PR #1239 created the same defect with inline-flex styles added to date-picker.

this focus inconsistent in previous version of date-picker was due to the date-picker being wrapped within terra-grid. To confirm these I modified terra-grid example to include date-picker and the focus style breaks as shown in screenshot : 
<img width="1558" alt="Screen Shot 2020-09-15 at 4 32 13 PM" src="https://user-images.githubusercontent.com/12869809/93204608-03c91c00-f774-11ea-962d-dcbc411d5780.png">
And again it was due to [flex styles](https://github.com/cerner/terra-core/blob/f03e926e522c5d7f9655f4f28bcee149fbbd1237/packages/terra-grid/src/Grid.module.scss#L13) that were used in grid causing the input to overlap the button on IE11. 

This is not just specific to terra-grid component even placing Date-Picker within a `div` having flex styles will create same issue in IE. 

So the changes made in PR #1248 will resolve this overlap issue of input in date-picker. 🎉
